src/spec/solver_spec.rb

```
@@ -0,0 +1,50 @@
+require_relative 'spec_helper'
+describe Solver do
+  describe '#factorial' do
+    it 'return 1 if parameter is 0' do
+      Solver.factorial(0).should eql 1
+    end
+  end
+
+  describe '#factorial' do
+    it 'return 24 if parameter is 4' do
+      Solver.factorial(4).should eql 24
+    end
+  end
+
+  describe '#factorial' do
+    it 'raise an exception if parameter is negative' do
+      expect { Solver.factorial(-1) }.to raise_error('Number is negative')
+    end
+  end
```

- [ ] This should raise an exception, kindly update it
```ruby
     ...
     describe Resolver do
         describe "#factorial" do
             it "raises exception for negative number" do
             resolver = Resolver.new
             expect { resolver.factorial(-5) }.to raise_exception
             end
         end
     end
```